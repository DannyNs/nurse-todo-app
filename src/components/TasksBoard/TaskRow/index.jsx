import React from 'react';
import PropType from 'prop-types';
import { observer } from 'mobx-react';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import SvgIcon from '@material-ui/core/SvgIcon';

import './index.scss';

const TaskRow = observer(({
  name, description, time, done, markAsDone,
}) => (
  <div className="nr-task-row">
    <Grid container spacing={24}>
      <Grid item xs={6}>
        {time}
      </Grid>
      <Grid item xs={6}>
        {time}
      </Grid>
      <Grid item xs={8}>
        <Grid item xs={12} className="nr-task-row__name">
          {name}
        </Grid>
        <Grid item xs={12}>
          {description}
        </Grid>
      </Grid>
      <Grid item xs={4}>
        {
          done ? (
            <SvgIcon className="nr-task-row__done">
              <path d="M19.77 5.03l1.4 1.4L8.43 19.17l-5.6-5.6 1.4-1.4 4.2 4.2L19.77 5.03m0-2.83L8.43 13.54l-4.2-4.2L0 13.57 8.43 22 24 6.43 19.77 2.2z" />
            </SvgIcon>) : (
              <Button variant="contained" onClick={markAsDone}>
                Done
              </Button>
          )
        }
      </Grid>
    </Grid>
  </div>
));

TaskRow.propTypes = {
  name: PropType.string.isRequired,
  description: PropType.string.isRequired,
  time: PropType.number.isRequired,
  markAsDone: PropType.func.isRequired,
};

export default TaskRow;
