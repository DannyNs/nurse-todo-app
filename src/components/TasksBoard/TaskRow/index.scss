.nr-task-row {
  
  &--right {
    text-align: right;
  }

  &__done {
    color: var(--main-success-color);
  }

  &__name {
    color: #000000;
    font-weight: bold;
  }
}