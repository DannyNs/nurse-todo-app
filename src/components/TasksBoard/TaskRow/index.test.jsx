import React from 'react';
import { shallow } from 'enzyme';
import TaskRow from '.';

describe('TaskRow', () => {
  it('should render without crashing', () => {
    // given
    const name = 'Mark Darcy';
    const description = 'Give pills';
    const time = 1241241241;
    const done = false;
    const markAsDone = () => {};

    // when
    const wrapper = shallow(
      <TaskRow
        name={name}
        description={description}
        time={time}
        done={done}
        markAsDone={markAsDone}
      />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
