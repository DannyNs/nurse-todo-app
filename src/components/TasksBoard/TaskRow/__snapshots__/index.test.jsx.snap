// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskRow should render without crashing 1`] = `
<div
  className="nr-task-row"
>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={6}
    >
      1970-01-15
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="nr-task-row--right"
      item={true}
      xs={6}
    >
      09:47:21
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={8}
    >
      <WithStyles(Grid)
        className="nr-task-row__name"
        item={true}
        xs={12}
      >
        Mark Darcy
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={12}
      >
        Give pills
      </WithStyles(Grid)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="nr-task-row--right"
      item={true}
      xs={4}
    >
      <WithStyles(Button)
        onClick={[Function]}
        variant="contained"
      >
        Done
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
